cmd/mdatagen/main.go

```diff
@@ -124,6 +124,23 @@ func run(ymlPath string) error {
 		toGenerate[filepath.Join(tmplDir, "resource.go.tmpl")] = filepath.Join(codeDir, "generated_resource.go")
 		toGenerate[filepath.Join(tmplDir, "resource_test.go.tmpl")] = filepath.Join(codeDir, "generated_resource_test.go")
 	}
+	if err = generateFile(filepath.Join(tmplDir, "match.go.tmpl"),
```

Comment:
This will be generated for every component in contrib. We need to generate these files only for scrapers `len(md.Metrics) > 0`.

Reply:
Done in [f416166](https://github.com/open-telemetry/opentelemetry-collector/pull/10065/commits/f4161668f1a6afea7486679a3d2d3f35495c5410)

Reply:
Correction in [a522523](https://github.com/open-telemetry/opentelemetry-collector/pull/10065/commits/a52252349053ff3d76313b6e476f75fa3d5bd423)